core/arch/arm/kernel/generic_boot.c

```diff
@@ -551,6 +554,44 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_secure_device_node(void *fdt)
+{
+	const uint32_t *p;
+	int offs;
+	int len;
+#if (TRACE_LEVEL >= TRACE_INFO)
+	char s[33];
+	int n = 0;
+	int m;
+#endif
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "secure-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(secure_device_id))
+		len = sizeof(secure_device_id);
+
+	memcpy(secure_device_id, p, len);
+	secure_device_id_len = len;
+
```

Comment:
It seems I did get your point... but you want to wipe all `secure-` guys IIUI.

This is the `make clean` thing again... if you want to do that nicely, you should have a node containing them and delete the node.

However this patch isn't about "all secure- properties".  The guy who adds the patch that hands this DTB on to NS world....(eh... it makes no sense to me... a-t-f gave the DTB to OP-TEE and then OP-TEE will give it back to a-t-f to give to Normal World?  That is not how it works atm... anyway...).

If out of consideration for this happy moment in the future, you want to move `secure-device-id` under a sacrificial, make-clean-style node, that'd make sense and I can adapt it.

Trying to get every user to clean his node though ends up with nodes with no current user in OP-TEE being left in etc... it's a pretty exact rerun of make clean... this isn't otherwise my problem I think...